packages/components/atoms/f-button/src/components/_tests/Button.test.js

```diff
@@ -5,11 +5,91 @@ describe('Button', () => {
     const actionType = 'button';
 
     it('should be defined', () => {
-        const propsData = { };
+        const propsData = {};
         const wrapper = shallowMount(FButton, { propsData });
         expect(wrapper.exists()).toBe(true);
     });
 
+    describe('methods ::', () => {
+        describe('validateProps ::', () => {
+            it.each([
+                'primary',
+                'secondary',
+                'outline',
+                'ghost',
+                'link'
+            ])('should not throw an error when the component is a standard button (isIcon="false") and buttonType is set to %s', buttonType => {
+                // Arrange
+                const propsData = {
+                    buttonType,
+                    isIcon: false
+                };
+
+                // Act
+                expect(() => {
+                    shallowMount(FButton, { propsData });
+                })
+                    // Assert
+                    .not.toThrowError();
+            });
+            it('should throw an error when the component is a standard button (isIcon="false") and buttonType is set to an invalid type', () => {
+                // Arrange
+                const spy = jest.spyOn(global.console, 'error').mockImplementation(() => { });
+                const propsData = {
+                    buttonType: 'invalid_value',
+                    isIcon: false
+                };
+
+                // Act
+                expect(() => {
+                    shallowMount(FButton, { propsData });
+                })
+                    // Assert
+                    .toThrowError('button is set to have buttonType="invalid_value"');
+
+                spy.mockRestore();
```

Comment:
You could move this into a `beforeEach`?

Reply:
Done